cpu/stm32/include/clk/cfg_clock_common2.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *               2017 OTA keys S.A.
+ *               2018-2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_stm32
+ * @{
+ *
+ * @file
+ * @brief       Base STM32F4 clock configuration
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef CLK_CFG_CLOCK_COMMON2_H
+#define CLK_CFG_CLOCK_COMMON2_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Clock system configuration (L0/L1/L4/L5/WB)
+ * @{
+ */
+/* Select the desired system clock source between PLL, HSE or HSI */
+#ifndef CONFIG_USE_CLOCK_PLL
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSE) || IS_ACTIVE(CONFIG_USE_CLOCK_HSI) || \
+    IS_ACTIVE(CONFIG_USE_CLOCK_MSI)
+#define CONFIG_USE_CLOCK_PLL            0
+#else
+#define CONFIG_USE_CLOCK_PLL            1     /* Use PLL by default */
+#endif
+#endif /* CONFIG_USE_CLOCK_PLL */
+
+#ifndef CONFIG_USE_CLOCK_MSI
+#define CONFIG_USE_CLOCK_MSI            0
+#endif /* CONFIG_USE_CLOCK_MSI */
+
+#ifndef CONFIG_USE_CLOCK_HSE
+#define CONFIG_USE_CLOCK_HSE            0
+#endif /* CONFIG_USE_CLOCK_HSE */
+
+#ifndef CONFIG_USE_CLOCK_HSI
+#define CONFIG_USE_CLOCK_HSI            0
+#endif /* CONFIG_USE_CLOCK_HSI */
```

Comment:
Is this needed? E.g. `IS_ACTIVE(CONFIG_USE_CLOCK_MSI)` would evaluate as false when undefined and

```C
#if CONFIG_USE_CLOCK_HSI
#endif
```
would also not trigger.

Reply:
That's a good point. I think I was biased by the initial way of handling this. I'll try.